test_parallelization.py fails on machines with many CPUs (e.g. EC2)

## Summary

`tests/infrastructure/test_parallelization.py::test_inner_parallelization_setup_in_workers` fails on EC2 instances (or any machine with more CPUs than outer splits) but passes in CI (GitHub Actions runners with 4 CPUs).

## Error

```
AssertionError: assert 9 == 1
```

at `tests/infrastructure/test_parallelization.py:46`:
```python
for lib in threadpool_info:
    assert lib["num_threads"] == 1
```

## Root Cause

The test asserts that all thread pools inside the worker have `num_threads == 1`. However, the production code in `ray_parallel.py:289` wraps `run_fn` with `threadpool_limits(limits=self.n_cpus)` where `self.n_cpus = cpus_per_worker`.

On a 4-CPU CI runner with 4 outer splits: `cpus_per_worker = 4 // 4 = 1`, so the assertion passes by coincidence.

On a 36-CPU EC2 instance with 4 outer splits: `cpus_per_worker = 36 // 4 = 9`, so `threadpool_limits(limits=9)` sets threads to 9, not 1, and the assertion fails.

## Environment

- EC2 instance with 36 CPUs
- Python 3.12
- Ray (local instance)
- The test passes on GitHub Actions `ubuntu-latest` (4 CPUs)

## Expected Fix

The assertion should be `assert lib["num_threads"] == n_cpus_per_worker` (or `== resources.cpus_per_worker`) since the outer worker is designed to allow `cpus_per_worker` threads for inner parallelization.

--- a/tests/infrastructure/test_parallelization.py
+++ b/tests/infrastructure/test_parallelization.py
@@ -43,7 +43,7 @@ def test_inner_parallelization_setup_in_workers(tmp_path, outer_splits):
         assert len(threadpool_info) >= 2
 
         for lib in threadpool_info:
-            assert lib["num_threads"] == 1
+            assert lib["num_threads"] == n_cpus_per_worker
 
     ray_parallel.run_parallel_outer(
         outer_split_data=outer_splits,

